### DISCUSSION TITLE
Cannot use PDF Preview with Latex Workshop: Tries to resolve some webview subdomain

### DISCUSSION THREAD

**Discussion by MexHigh** [eclipse-theia/theia#7988] [General]
### Bug Description:
I have Latex Workshop installed and working perfectly, only the webview does not work. It tries to resolve to a `<uuid>.webview.<theia host>` subdomain which i don't have A or AAAA records for.
This does not only apply to Latex previews, but for normal PDF files too (as long as I don't rightclick --> open with --> preview. This does work and opens my browsers default pdf viewer.)

![Screenshot](https://user-images.githubusercontent.com/50419884/84270859-a0ec3700-ab2b-11ea-9278-ce3da2de4730.png)

Just removed some personal information :)
I'm hosting Theia on my own infrastructure for private use. I'm using the caddy webserver as a tls reverse proxy but this error occures even if i bypass it.

### Steps to Reproduce:

1. Installed Latex Workshop via vsix
2. Opened a .tex file
3. Clicked on the preview button over the editor on the right
4. Choose "in VSCode tab"

### Additional Information

- Operating System: Ubuntu Server LTS 18.04
- Theia Version: In a selfmade Docker container (parent image: `ubuntu:18.04`, using the same package.json as in https://github.com/theia-ide/theia-apps/blob/master/theia-full-docker/latest.package.json, only added latex-workshop to the `theiaPlugins` key, but this error occures without the change with normale PDFs as well)

What do i have to do to suppress theia to use those generated subdomains? I've read about the `THEIA_WEBVIEW_EXTERNAL_ENDPOINT` envvar but I don't quite get what it is used for. If i set it to the same domain theia is running, the webview tries to resolve to `localhost` with some high port instead which does not work either.

**Comment by yuhorun** [comment]
> 
> ### Bug Description:
> I have Latex Workshop installed and working perfectly, only the webview does not work. It tries to resolve to a `<uuid>.webview.<theia host>` subdomain which i don't have A or AAAA records for.
> This does not only apply to Latex previews, but for normal PDF files too (as long as I don't rightclick --> open with --> preview. This does work and opens my browsers default pdf viewer.)
> 
> ![Screenshot](https://user-images.githubusercontent.com/50419884/84270859-a0ec3700-ab2b-11ea-9278-ce3da2de4730.png)
> 
> Just removed some personal information :)
> I'm hosting Theia on my own infrastructure for private use. I'm using the caddy webserver as a tls reverse proxy but this error occures even if i bypass it.
> 
> 
> ### Steps to Reproduce:
> 
> 1. Installed Latex Workshop via vsix
> 2. Opened a .tex file
> 3. Clicked on the preview button over the editor on the right
> 4. Choose "in VSCode tab"
> 
> 
> 
> 
> ### Additional Information
> 
> - Operating System: Ubuntu Server LTS 18.04
> - Theia Version: In a selfmade Docker container (parent image: `ubuntu:18.04`, using the same package.json as in https://github.com/theia-ide/theia-apps/blob/master/theia-full-docker/latest.package.json, only added latex-workshop to the `theiaPlugins` key, but this error occures without the change with normale PDFs as well)
> 
> 
> What do i have to do to suppress theia to use those generated subdomains? I've read about the `THEIA_WEBVIEW_EXTERNAL_ENDPOINT` envvar but I don't quite get what it is used for. If i set it to the same domain theia is running, the webview tries to resolve to `localhost` with some high port instead which does not work either.


I also encountered this problem, have you made any progress?
